src/app/counselor/register/page.jsx

```diff
@@ -0,0 +1,114 @@
+import { useApiMutation } from '~/app/Lib/apiFetch'
+import { buhosLogo } from '~/app/images'
+
+const displayName = 'CounselorRegister'
+const classes = getClassPrefixer(displayName)
+
+const Container = styled('div')(({ theme }) => ({
+  display: 'flex',
+  justifyContent: 'center',
+  alignItems: 'center',
+  height: '100vh',
+  width: '100vw',
+  backgroundColor: theme.palette.primary.main,
+  [`& .${classes.contentBox}`]: {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: '1rem',
+    justifyContent: 'center',
+    alignItems: 'center',
+    width: '420px',
+    height: '550px',
+    backgroundColor: theme.palette.contrast.main,  
+    borderRadius: 4,
+    padding: '1rem',
+    textAlign: 'center'
+  },
+}))
+
+const CounselorRegister = ({ snackbarMessage, setSnackbarMessage }) => {
+  return(
+    <Container>
+      <div className={classes.contentBox}>
+        <Image src={buhosLogo} width={270} height={200} alt='Owls Logo'/>
+        <Typography variant="h5">Bienvenido a OwlsHub</Typography>
+        <Stack spacing={1} width="90%">
+          <Field 
+            component={TextField} 
+            fullWidth name="names" 
+            placeholder="Nombre"/>
+          <Field 
+            component={TextField} 
+            fullWidth name="password" 
+            type="password" 
+            placeholder="Contraseña"/>
+          <Field 
+            component={TextField} 
+            fullWidth 
+            name="repeatPass" 
+            type="password" 
+            placeholder="Repetir contraseña"/>
+        </Stack>
+        <Form>
+          <Button type="submit" variant="contained" >Enviar</Button>
+        </Form>
+        <Snackbar 
+          open={Boolean(snackbarMessage)}
+          autoHideDuration={4000}
+          onClose={() => setSnackbarMessage(null)}
+          anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
+          message={snackbarMessage}/>
+      </div>
+    </Container>
+  )
+}
+
+const Wrapper = () => {
+  const [snackBarMessage, setSnackbarMessage] = useState(null)
+  const register = useApiMutation({ path: 'me', opts: { method: 'PATCH' } })
+  const router = useRouter()
+  const initialValues = getRegisterInitialValues()
+  const validationSchema = getRegisterValidationSchema()
+
+  const handleSubmit = async values => {
+    const payload = omit(['repeatPass'], values)
+    await register.mutate(payload, {
+      onSuccess: () => {
+        setSnackbarMessage('Registro exitoso')
+        setTimeout(() => {
+          router.replace('/counselor')
+        },2000)
+      },
+      onError: () => {
+        setSnackbarMessage('Lo sentimos, ha ocurrido un error')
+      }
+    })
+  }
+  return (
+    <Formik
+      initialValues={initialValues}
+      validationSchema={validationSchema}
+      onSubmit={handleSubmit}> 
+      <CounselorRegister 
+        snackbarMessage={snackBarMessage}
+        setSnackbarMessage={setSnackbarMessage}/>
+    </Formik>
```

Comment:
Aquí es importante que protejas esta ruta, ya que solo deberían poder acceder usuarios con ciertos permisos específicos.

Te recomiendo revisar la carpeta `Permissions` del proyecto de la OMICHH. Ahí puedes ver cómo se están implementando las validaciones de permisos, y podrías reutilizar esa lógica para asegurar que solo usuarios autorizados accedan a esta parte del sistema.

Además, asegúrate de **cerrar correctamente los tags** que estés utilizando. Por ejemplo:

```suggestion
    <Formik
      initialValues={initialValues}
      validationSchema={validationSchema}
      onSubmit={handleSubmit}
    > 
      <CounselorRegister 
        snackbarMessage={snackBarMessage}
        setSnackbarMessage={setSnackbarMessage}
      />
    </Formik>
```